File: 342-power-of-four/342-power-of-four.js
/**
 * @param {number} n
 * @return {boolean}
 */
var isPowerOfFour = function(n) {
   if (n === 0) return false;
    if (n === 1) return true;
    
    let power = 1;
    while (power < n) {
        power = power * 4
    }
    if (power === n)return true;
    else return false; 
};

File: 342-power-of-four/342-power-of-four.rb
# @param {Integer} n
# @return {Boolean}
def is_power_of_four(n)
    if n === 0 
        false 
    end 
    if n === 1
        true 
    end 
    
    power = 1
    while power < n 
        power = power * 4 
    end 
    
    if power === n 
        true 
    else 
        false
    end 
end